File: GeneratedEntityReaderClasses/Components/ExplosionComponent.cs
namespace GeneratedNoitaClasses;
public class ExplosionComponent : NoitaComponentBase,  iNoitaType<ExplosionComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaUInt trigger { get; set; } = new();
	public ConfigExplosion config_explosion { get; set; } = new();
	public NoitaInt timeout_frames { get; set; } = new();
	public NoitaInt timeout_frames_random { get; set; } = new();
	public NoitaBool kill_entity { get; set; } = new();
	public NoitaInt mTimerTriggerFrame { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		trigger.Read(s);
		config_explosion.Read(s);
		timeout_frames.Read(s);
		timeout_frames_random.Read(s);
		kill_entity.Read(s);
		mTimerTriggerFrame.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		trigger.Write(s);
		config_explosion.Write(s);
		timeout_frames.Write(s);
		timeout_frames_random.Write(s);
		kill_entity.Write(s);
		mTimerTriggerFrame.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/ExplosionComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaUInt
from GeneratedClasses.ConfigExplosion import ConfigExplosion
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class ExplosionComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    trigger:NoitaUInt = field(default_factory=NoitaUInt)
    config_explosion:ConfigExplosion = field(default_factory=ConfigExplosion)
    timeout_frames:NoitaInt = field(default_factory=NoitaInt)
    timeout_frames_random:NoitaInt = field(default_factory=NoitaInt)
    kill_entity:NoitaBool = field(default_factory=NoitaBool)
    mTimerTriggerFrame:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.trigger.Read(s)
        self.config_explosion.Read(s)
        self.timeout_frames.Read(s)
        self.timeout_frames_random.Read(s)
        self.kill_entity.Read(s)
        self.mTimerTriggerFrame.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.trigger.Write(s)
        self.config_explosion.Write(s)
        self.timeout_frames.Write(s)
        self.timeout_frames_random.Write(s)
        self.kill_entity.Write(s)
        self.mTimerTriggerFrame.Write(s)

